core/sql/optimizer/SynthType.cpp

```diff
@@ -5207,6 +5207,24 @@ const NAType *Translate::synthesizeType()
          err4106arg = SQLCHARSETSTRING_UTF8;
        break;
 
+     case GBK_TO_UTF8:
+       if (translateSource->getCharSet() == CharInfo::GBK )
+         charsetTarget = CharInfo::UTF8;
+       else
+       {
+           /* this is a solution to support GBK before Trafodion can support GBK in total
+            * see jira 1720 for more details
+            * the logic here is:
+            * when HIVE_FILE_CHARSET is not empty, it means the real charset in Hive table is not same as HIVE_DEFAULT_CHARSET
+            * in this case, allow the converting , ignoring the source charset checking above
+            */
+            if( CmpCommon::getDefaultString(HIVE_FILE_CHARSET) ==  "" ) //CmpCommon::getDefaultString(HIVE_DEFAULT_CHARSET) )
```

Comment:
See my other comment about where to put the automatic conversion. If you do that, this check will not be needed. Otherwise, this logic will now allow a TRANSLATE(GBKTOUTF8) anywhere in SQL statements. Allowing this just gives people an opportunity to shoot themselves in the foot, so it may be better to avoid it.